src/goesproc/proj.cc

```diff
@@ -5,56 +5,35 @@
 
 namespace {
 
-std::vector<std::string> toVector(
-  const std::map<std::string, std::string>& args) {
-  std::vector<std::string> vargs;
-  vargs.reserve(args.size());
-  for (const auto& arg : args) {
-    std::stringstream ss;
-    ss << arg.first << "=" << arg.second;
-    vargs.push_back(ss.str());
-  }
-  return vargs;
-}
-
 std::string pj_error(std::string prefix = "proj: ") {
   std::stringstream ss;
-  ss << prefix << pj_strerrno(pj_errno);
+  ss << prefix << proj_errno_string(proj_errno(NULL));
   return ss.str();
 }
 
 } // namespace
 
-Proj::Proj(const std::vector<std::string>& args) {
-  std::vector<char*> argv;
-  for (const auto& arg : args) {
-    argv.push_back(strdup(arg.c_str()));
-  }
-  proj_ = pj_init(argv.size(), argv.data());
+Proj::Proj(const std::string& args) {
+  proj_ = proj_create(NULL, args.c_str());
   if (!proj_) {
     throw std::runtime_error(pj_error("proj initialization error: "));
   }
-  for (auto& arg : argv) {
-    free(arg);
-  }
-}
-
-Proj::Proj(const std::map<std::string, std::string>& args)
-  : Proj(toVector(args)) {
 }
 
 Proj::~Proj() {
-  pj_free(proj_);
+  proj_destroy(proj_);
 }
 
 std::tuple<double, double> Proj::fwd(double lon, double lat) {
-  projUV in = { lon, lat };
-  projXY out = pj_fwd(in, proj_);
-  return std::make_tuple<double, double>(std::move(out.u), std::move(out.v));
+  PJ_COORD in;
+  in.uv = { lon, lat };
+  PJ_COORD out = proj_trans(proj_, PJ_FWD, in);
+  return std::make_tuple<double, double>(std::move(out.xy.x), std::move(out.xy.y));
```

Comment:
Nit: the previous version of this line references `out.u` and `out.v`, but I think that these should be `x` and `y`?  The change is a no-op because of the underlying union, but I think `x` and `y` is more correct?